### ISSUE TITLE
Kubernetes Audit Policy - Exclude SCK Logging user

### ISSUE THREAD

**Issue by brianp9906** [splunk/splunk-connect-for-kubernetes#370]
Is anyone familiar with Kubernetes audit policy with respect to SCK logging agent exclusions?

The audit policy is being spammed for "watch" on pods by the SCK service account.  I tried setting log level to "None" but doesnt seem to help.  Any suggestions?

I started with the audit policy here:
https://www.outcoldsolutions.com/docs/monitoring-kubernetes/v5/audit/

Then changed 
system:serviceaccount:collectorforkubernetes:collectorforkubernetes
to
system:serviceaccount:splunk:sck-14-splunk-kubernetes-logging

sample audit log entry that is spamming. 

`{"kind":"Event","apiVersion":"audit.k8s.io/v1","level":"Request","auditID":"f79f2847-988c-4e0e-8573-341ac661a7b6","stage":"ResponseComplete","requestURI":"/api/v1/watch/pods?fieldSelector=spec.nodeName%3Dworkernode3\u0026resourceVersion=20173104","verb":"watch","user":{"username":"system:serviceaccount:splunk:sck-14-splunk-kubernetes-logging","uid":"85d07c9f-0a8a-41a2-b13a-ab25537c460a","groups":["system:serviceaccounts","system:serviceaccounts:splunk","system:authenticated"]},"sourceIPs":["10.XXX.XXX.XXX"],"userAgent":"http.rb/4.3.0","objectRef":{"resource":"pods","apiVersion":"v1"},"responseStatus":{"metadata":{},"status":"Success","message":"Connection closed early","code":200},"requestReceivedTimestamp":"2020-04-16T18:46:42.669041Z","stageTimestamp":"2020-04-16T18:46:42.669410Z","annotations":{"authorization.k8s.io/decision":"allow","authorization.k8s.io/reason":"RBAC: allowed by ClusterRoleBinding \"sck-14-splunk-kubernetes-logging\" of ClusterRole \"sck-14-splunk-kubernetes-logging\" to ServiceAccount \"sck-14-splunk-kubernetes-logging/splunk\""}}`

I thought this would skip the audit logging for SCK service account but I guess not?

  # Do not log from collector
  - level: None
    users:
    - system:serviceaccount:splunk:sck-14-splunk-kubernetes-logging

**Comment by rockb1017** [comment]
which cluster provider are you on?

**Comment by rockb1017** [comment]
this seems similar to other issue we are releasing the fix for. 
Could you use this image instead for logging pod ?
`rock1017/fluentd-hec:test2`

**Comment by brianp9906** [comment]
I’m confused how a new image affects the audit policy exclusion?

**Comment by matthewmodestino** [comment]
Hi @brianp9906!

If I understand the Kubernetes docs correctly, I believe we need to add something like:

```
  # Don't log watch requests by the "system:kube-proxy" on endpoints or services
  - level: None
    users: ["system:kube-proxy"]
    verbs: ["watch"]
    resources:
    - group: "" # core API group
      resources: ["endpoints", "services"]
```
where we say dont log watch requests from connect for k8s service accounts for pods and namespaces. 

https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy

**Comment by brianp9906** [comment]
I tried so many different audit policy variations that it was driving me nuts.  The documentation from kubernetes.io does not state anything about what to do after to apply it and more importantly make it take effect.   
I finally just killed kube-apiserver process to make kubernetes restart the kube-api server and that forced a re-read of the audit config.  Using Systemctl (redhat) to restart kubelet didnt apply the policy yaml file correctly.
